src/Raven.Server/Web/System/AdminMonitoringHandler.cs

```diff
@@ -0,0 +1,591 @@
+// -----------------------------------------------------------------------
+//  <copyright file="AdminMonitoringHandler.cs" company="Hibernating Rhinos LTD">
+//      Copyright (c) Hibernating Rhinos LTD. All rights reserved.
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Linq;
+using Raven.Client.Util;
+using Raven.Server.Documents;
+using Raven.Server.Documents.Indexes;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils;
+using Raven.Server.Utils.Monitoring;
+using Sparrow;
+using Sparrow.Binary;
+using Sparrow.Json;
+using Sparrow.LowMemory;
+using Sparrow.Server.Extensions;
+using Sparrow.Server.LowMemory;
+using Sparrow.Server.Utils;
+using Voron;
+using Index = Raven.Server.Documents.Indexes.Index;
+using Size = Sparrow.Size;
+
+namespace Raven.Server.Web.System
+{
+    public class AdminMonitoringHandler : RequestHandler
+    {
+        [RavenAction("/admin/monitoring/v1/server", "GET", AuthorizationStatus.Operator)]
+        public Task MonitoringServer()
+        {
+            AssertMonitoring();
+
+            var result = new ServerMetrics();
+
+            result.ServerVersion = ServerWide.ServerVersion.Version;
+            result.ServerFullVersion = ServerWide.ServerVersion.FullVersion;
+            result.UpTimeInSec = (int)Server.Statistics.UpTime.TotalSeconds;
+
+            using (var currentProcess = Process.GetCurrentProcess())
+                result.ServerProcessId = currentProcess.Id;
+
+            result.Config = GetConfigMetrics();
+            result.Backup = GetBackupMetrics();
+            result.Cpu = GetCpuMetrics();
+            result.Memory = GetMemoryMetrics();
+            result.Network = GetNetworkMetrics();
+            result.License = GetLicenseMetrics();
+            result.Disk = GetDiskMetrics();
+            result.Certificate = GetCertificateMetrics();
+            result.Cluster = GetClusterMetrics();
+            result.Databases = GetAllDatabasesMetrics();
```

Comment:
I'm a bit worried that there is no ability to say what should be returned. Can we add switches e.g. `?cluster=false` ?

Reply:
collecting all data is O(1) so I don't think we should worry about that. I just grouped them into categories.